central/image/service/service_impl.go

```diff
@@ -203,14 +208,48 @@ func internalScanRespFromImage(img *storage.Image) *v1.ScanImageInternalResponse
 	}
 }
 
-func (s *serviceImpl) saveImage(img *storage.Image) error {
+func (s *serviceImpl) saveImage(ctx context.Context, img *storage.Image) error {
+	if features.ScannerV4Enabled.Enabled() && !scannedByScannerV4(img) && s.scannedByClairify(img) {
+		// This image was scanned by the old Clairify scanner, we do not want to
+		// overwrite an existing Scanner V4 scan in the database (if it exists).
+		if existingImg, exists, err := s.datastore.GetImage(ctx, img.GetId()); err != nil {
+			return err
+		} else if exists && scannedByScannerV4(existingImg) {
+			// Note: This image will not have `RiskScore` fields populated because
+			// risk scores are heavily tied to upserting into Central DB and
+			// this image is not being upserted.
+			log.Warnw("Cannot overwrite Scanner V4 scan already in DB with Clairify scan and cannot calculate risk scores", logging.ImageName(img.GetName().GetFullName()))
+			return nil
+		}
```

Comment:
nit to remove the `else`

```suggestion
		existingImg, exists, err := s.datastore.GetImage(ctx, img.GetId())
		if err != nil {
			return err
		} 
		if exists && scannedByScannerV4(existingImg) {
			// Note: This image will not have `RiskScore` fields populated because
			// risk scores are heavily tied to upserting into Central DB and
			// this image is not being upserted.
			log.Warnw("Cannot overwrite Scanner V4 scan already in DB with Clairify scan and cannot calculate risk scores", logging.ImageName(img.GetName().GetFullName()))
			return nil
		}
```

Reply:
Perhaps this is a good candidate for the style guide? My thought was to minimally scope the variables, in this case to the conditions.